bee-common/bee-packable/src/packable/mod.rs

```diff
@@ -61,11 +61,23 @@ pub trait Packable: Sized {
     }
 
     /// Unpacks this value from the given [`Unpacker`].
-    fn unpack<U: Unpacker>(unpacker: &mut U) -> Result<Self, UnpackError<Self::UnpackError, U::Error>>;
+    fn unpack<U: Unpacker, const CHECK: bool>(
```

Comment:
I'm not sure if we should expose `CHECK` in a public trait interface, because it feels like an implementation detail. Wouldn't `unpack_checked` and `unpack_unchecked` suffice, while keeping `unpack<..,const CHECK: bool>` internally?

Reply:
Hmmm... I agree that having `CHECK` exposed at the same time as `unpack_(un)?checked` is redundant but those last are supposed to be just helper methods like `pack_to_vec`. 

The other thing is that I'm not sure how to keep `CHECK` hidden but at the same time allow the user to implement it.

Reply:
Would most users use `unpack_(un)?checked` or `unpack<U: Unpacker, const CHECK: bool>` in their code? As far as I understand, the purpose of the latter is to only have one method that the users of the trait have to implement?

Reply:
The older API is `unpack<R: Read, const CHECK: bool>` (in cpt2/dev), so that may be more familiar to existing users.

I guess another thing to consider is because `unpack_unchecked` and `unpack_checked` aren't hidden, a user could effectively implement all three, with three different behaviours, which doesn't make sense. While that situation isn't likely, and I'm not sure whether it actually needs considering, it might make more sense to either offer `unpack`, or `unpack_unchecked` and `unpack_checked`, rather than all three.

Reply:
Maybe we can do something similar to what we did with the `coerce` module and create a `PackableExt` trait that is only implemented for `P: Packable` and seal it so the user cannot override its behaviour. What say you?